File: 231.power-of-two.cs
/*
 * @lc app=leetcode id=231 lang=csharp
 *
 * [231] Power of Two
 */

// @lc code=start
public class Solution {
    public bool IsPowerOfTwo(int n) {
        if (n == 2 || n == 1) {
            return true;
        } else if (n < 1 || n % 2 != 0) {
            return false;
        } else {
            return IsPowerOfTwo(n / 2);
        }
    }
}

// @lc code=end

File: 231.power-of-two.py
#
# @lc app=leetcode id=231 lang=python3
#
# [231] Power of Two
#

# @lc code=start
class Solution:
    def isPowerOfTwo(self, n: int) -> bool:
        if n == 2 or n == 1:
            return True
        elif n < 1 or n % 2:
            return False
        else:
            return self.isPowerOfTwo(n // 2)
# @lc code=end

